python/cudf/dataframe/dataframe.py

```diff
@@ -408,6 +409,127 @@ def __repr__(self):
             len(self),
         )
 
+    # binary, rbinary, unary, orderedcompare, unorderedcompare
+    def _call_op(self, other, internal_fn, fn):
+        result = DataFrame()
+        result.set_index(self.index)
+        if isinstance(other, Sequence):
+            for k, col in enumerate(self._cols):
+                result[col] = getattr(self._cols[col], internal_fn)(
+                        other[k],
+                        fn,
+                )
+        elif isinstance(other, DataFrame):
+            for col in other._cols:
+                if col in self._cols:
+                    result[col] = getattr(self._cols[col], internal_fn)(
+                            other._cols[col],
+                            fn,
+                    )
+                else:
+                    result[col] = Series(cudautils.full(self.shape[0],
+                                         np.dtype('float64').type(np.nan),
+                                         'float64'), nan_as_null=False)
+            for col in self._cols:
+                if col not in other._cols:
+                    result[col] = Series(cudautils.full(self.shape[0],
+                                         np.dtype('float64').type(np.nan),
+                                         'float64'), nan_as_null=False)
+        elif isinstance(other, Series):
+            raise NotImplementedError(
+                    "Series to DataFrame arithmetic not supported "
+                    "until strings can be used as indices. Try converting your"
+                    " Series into a DataFrame first.")
+        elif isinstance(other, numbers.Number):
+            for col in self._cols:
+                result[col] = getattr(self._cols[col], internal_fn)(
+                        other,
+                        fn,
+                )
+        else:
```

Comment:
What would trigger this else clause?

Reply:
I'm going to go with "nothing we support". I don't know why that is in there, a restructuring of the earlier clauses.

Reply:
See my fix.